src/main/java/io/javalin/plugin/openapi/dsl/openApiBuilderDsl.kt

```diff
@@ -110,9 +111,12 @@ fun Operation.applyMetaInfo(options: CreateSchemaOptions, path: PathParser, meta
         }
     }
 
-    if (documentation.hasFormParameter()) {
+    if (documentation.hasFormParameter() || documentation.hasFileUploads()) {
         updateRequestBody {
-            applyDocumentedFormParameters(documentation.formParameterList)
+            // Requests with file uploads need to be a multipart content
+            // Regular forms alone will be url encoded by default
+            val contentType = if (documentation.hasFileUploads()) ContentType.FORM_DATA_MULTIPART else ContentType.FORM_DATA_URL_ENCODED
```

Comment:
It feels better to get contentType inside applyDocumentedFormParameters, why do you need it here?

Reply:
Just a remnant from the first version - fixed it (y)